### PR TITLE
front: use reusable getDocument function

### LINKED ISSUE #3968: front: use getDocument everywhere
### Description and goal

Use the reusable getDocument function everywhere where it is needed.

### Acceptance criteria

The document is get and post only by the common functions

### PR DESCRIPTION
**Pull request by clarani**
closes #3968

I only changed the getDocument calls, not the other ones of the files I touched :)

### CODE REVIEW

**Review by SharglutDev — COMMENTED** [review]
Looks good to me :)
(one more file on my [PR](https://github.com/DGEXSolutions/osrd/pull/3965) will also use your getDocument function)

Just maybe in the project modal, you could also change the call in the update function ? (line 105)

File: front/src/applications/operationalStudies/components/Project/AddOrEditProjectModal.tsx [hunk 1]
```diff
@@ -102,7 +95,7 @@ export default function AddOrEditProjectModal({ editionMode, project, getProject
       try {
         let imageId = currentProject.image;
         if (tempProjectImage) {
-          imageId = await getDocKey(tempProjectImage as Blob);
+          imageId = await postDocument(tempProjectImage as Blob);
         } else if (tempProjectImage === null) {
           imageId = null;
           setTempProjectImage(imageId);
```
Comment (Yohh) [inline front/src/applications/operationalStudies/components/Project/AddOrEditProjectModal.tsx:123]: the result is:
image
maybe we should not talk about image**Id** ?

Comment (clarani) [reply front/src/applications/operationalStudies/components/Project/AddOrEditProjectModal.tsx:123]: No, you are looking at the response of the GET /document/, not the one of POST /document/ :) 
image

Comment (Yohh) [reply front/src/applications/operationalStudies/components/Project/AddOrEditProjectModal.tsx:123]: I was looking at tempProjectImage, I thought `setTempProjectImage(null)` could be better for good understanding but it's ok

**Review by alexandredamiron — APPROVED** [review]

**Pull request merged**